### ISSUE TITLE
ext.extlinks: Cannot escape angle brackets in link caption

### ISSUE THREAD

**Issue by theodelrieu** [sphinx-doc/sphinx#6375]
Hello,

I've posted a message on [sphinx-users](https://groups.google.com/forum/#!topic/sphinx-users/Zcbw_5lwU0A) Google Group, but it looks more like a bug to me, so I'll repost it here.

-----

Hello, I'm currently documenting a C++ API using Concepts, using Sphinx 2.0.1.

I have defined the following in my conf.py:

```python
extlinks = {
    'iterconcept': ('https://en.cppreference.com/w/cpp/experimental/ranges/iterator/%s', 'std::'),
}
```

This works fine for concepts with single template parameter:

>:iterconcept:\`RandomAccessIterator\`

However, when there is more than one parameter, it fails horribly:

>:iterconcept:\`Sentinel\<I\> \<Sentinel\>\`

The output is `std::Sentinel` (instead of `std::Sentinel<I>`), and the [link](https://en.cppreference.com/w/cpp/experimental/ranges/iterator/I%3E%20%3CSentinel) contains some invalid characters.

I tried to escape the first angle brackets with backslashes, with no success.

My goal is to get the correct link with `std::Sentinel<I>` rendered with the following [link](https://en.cppreference.com/w/cpp/experimental/ranges/iterator/Sentinel).

Am I missing something obvious?

**Comment by theodelrieu** [comment]
Note that the following (using vanilla hyperlinks, no extension) works as expected:

>\`google\<I\> \<https://www.google.com>`

The angle brackets are rendered, and the link is correct.

**Comment by theodelrieu** [comment]
There seems to be the same issue with `:doc:`. However, escaping the left angle brackets workaround the issue.

**Comment by tk0miya** [comment]
Fixed by #6376.
Thank you for reporting.
